readthedocs/core/views/serve.py

```diff
@@ -269,3 +272,74 @@ def robots_txt(request, project):
         return HttpResponse(open(fullpath).read(), content_type='text/plain')
 
     return HttpResponse('User-agent: *\nAllow: /\n', content_type='text/plain')
+
+
+@map_project_slug
+# TODO: make this cache dependent on the project's slug
+@cache_page(60 * 60 * 24 * 3)  # 3 days
+def sitemap_xml(request, project):
+
+    def priorities_generator():
+        """
+        Generator returning ``priority`` needed by sitemap.xml.
+
+        It generates values from 1 to 0.1 by decreasing in 0.1 on each
+        iteration. After 0.1 is reached, it will keep returning 0.1.
+        """
+        priorities = [1, 0.9, 0.8, 0.7, 0.6, 0.5, 0.4, 0.3, 0.2]
+        for p in priorities:
+            yield p
+
+        while True:
+            yield 0.1
+
+    def changefreqs_generator():
+        """
+        Generator returning ``changefreq`` needed by sitemap.xml.
+
+        It returns ``daily`` on first iteration, then ``weekly`` and then it
+        will return always ``monthly``.
+        """
+        changefreqs = ['daily', 'weekly']
+        for c in changefreqs:
+            yield c
+
+        while True:
+            yield 'monthly'
+
+    sorted_versions = sort_version_aware(project.versions.filter(active=True))
+
+    versions = []
+    for version, priority, changefreq in zip(
+            sorted_versions, priorities_generator(), changefreqs_generator()):
+        element = {
+            'loc': version.get_subdomain_url(),
```

Comment:
The URL should be properly escaped: https://www.sitemaps.org/protocol.html#escaping

Reply:
I'm not sure if there is something we need to do here, actually.